### PR TITLE
WIP: generate documentation for config maps

### PR DESCRIPTION
**Pull request by prashbnair**
Changes for generating documentation for the user facing config maps 

* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Review by simonpasquier — COMMENTED** [review]
Nice! From a presentation perspective I would expect to see 2 entries:
* one for `openshift-monitoring/cluster-monitoring-config`
* one for `openshift-user-workload-monitoring/user-workload-monitoring-config`

And if possible it would be neat to represent the YAML structure:

<pre>
prometheusK8s: # Configuration for prometheus
  logLevel: "" # One of debug, info, warn or error
  resources: {} # <a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#resourcerequirements-v1-core">*v1.ResourceRequirements</a>
  ...
</pre>

code source of :arrow_up::

```
<pre>
prometheusK8s: # Configuration for prometheus
  logLevel: "" # One of debug, info, warn or error
  resources: {} # <a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#resourcerequirements-v1-core">*v1.ResourceRequirements</a>
  ...
</pre>
```

A bit like https://thanos.io/v0.23/components/rule.md/#alertmanager.

**Comment by simonpasquier** [comment]
And I guess we need a job to verify that the generated documentation commited the repository doesn't drift.

File: Documentation/config.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
+---
```
Comment (arajkumar) [inline Documentation/config.md:1]: Is both config.md and config.adoc are valid? I see that only config.adoc being generated.

File: Documentation/config.adoc [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+// NOTE: The contents of this file are automatically generated from source code comments.
+// If you wish to make a change or an addition to the content in this document, do so by changing the code comments.
+
+= Config map fields for the Cluster Monitoring Operator 
+
+The following tables describe fields that you can use in config map objects for the Cluster Monitoring Operator. These fields enable you to do fine-grained configuration of platform monitoring and user workload monitoring.
+
+== ClusterMonitoringConfiguration
+
+
```
Comment (arajkumar) [inline Documentation/config.adoc:11]: I couldn't find `enableUserWorkload` in the generated doc., Am I missing something here?

**Pull request closed without merging**